src/DoctrineObject.php

```diff
@@ -110,6 +111,21 @@ public function setDefaultByReferenceStrategy($defaultByReferenceStrategy)
         return $this;
     }
 
+    /**
+     * Get all field names, this includes direct field names, names of embeddables and
+     * associations. By using a key-based generator, duplicates are effectively removed.
+     */
+    public function getFieldNames(): iterable
+    {
+        $fields = array_merge($this->metadata->getFieldNames(), $this->metadata->getAssociationNames());
+        foreach ($fields as $fieldName) {
+            if ($pos = strpos($fieldName, '.')) {
+                $fieldName = substr($fieldName, 0, $pos);
+            }
+            yield $fieldName => $fieldName;
```

Comment:
We don't need the same information twice, do we?
```suggestion
            yield $fieldName;
```